sqlitedict.py

```diff
@@ -282,13 +282,16 @@ def close(self, do_log=True):
         if do_log:
             logger.debug("closing %s" % self)
         if hasattr(self, 'conn') and self.conn is not None:
-            if self.conn.autocommit:
-                # typically calls to commit are non-blocking when autocommit is
-                # used.  However, we need to block on close() to ensure any
-                # awaiting exceptions are handled and that all data is
-                # persisted to disk before returning.
-                self.conn.commit(blocking=True)
-            self.conn.close()
+            try:
+                if self.conn.autocommit:
+                    # typically calls to commit are non-blocking when
+                    # autocommit is used.  However, we need to block on close()
+                    # to ensure any awaiting exceptions are handled and that
+                    # all data is persisted to disk before returning.
+                    self.conn.commit(blocking=True)
+                self.conn.close()
+            except Exception:
+                pass
```

Comment:
I think @tmylk 's is a good point -- this "exception check" belongs to `__del__` then, not to `close`.

If `close` fails under normal circumstances (not during process exit), we don't want to cover and silence that.